google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -482,12 +482,19 @@ def delete table, *id
         #     end
         #   end
         #
-        def transaction
+        def transaction &block
           ensure_service!
           tx_session = session
           tx_grpc = @project.service.begin_transaction tx_session.path
           tx = Transaction.from_grpc(tx_grpc, tx_session)
-          yield tx
+          begin
+            block.call tx
+          rescue Google::Cloud::AbortedError
+            # Retry the entire transaction
```

Comment:
I also left this out for now. I haven't been able to observe an ABORTED response from the API, so I don't know for sure how the value will be returned. Also, the grpc client is currently fixing some issues with exposing the trailing metadata.